src/software/thunderscope/robot_communication.py

```diff
@@ -113,7 +117,7 @@ def setup_for_fullsystem(self) -> None:
 
         self.receive_ssl_referee_proto = tbots_cpp.SSLRefereeProtoListener(
             SSL_REFEREE_ADDRESS,
-            SSL_REFEREE_PORT,
+            self.referee_port,
             lambda data: self.current_proto_unix_io.send_proto(Referee, data),
             True,
         )
```

Comment:
One thing that I would imagine we have to take into account, is the fact that `GameController` class also starts a `SSLRefereeProtoListener`, meaning that we would have two listeners on the same port, which is not good.

Also, per my last comment, `SSL_REFEREE_PORT` is 10003, which is the actual conventional port. We should make sure we're listening on that port whenever we're not launching our own GC.

Reply:
I have a feeling that this is solve by a default parameter.